### ISSUE TITLE
Gdn_ApplicationManager::testApplication() DROPS plugin data!

### ISSUE THREAD

**Issue by mtschirs** [vanilla/vanilla#2931]
Gdn_ApplicationManager::testApplication() calls the respective application's setup method but fails to take structure changes made by plugins into account.

As a result, if a plugin has data stored in an additional column added to the application's table, this data is deleted!

Due to this flaw, the mechanism in https://github.com/vanilla/vanilla/blob/fd5346/applications/dashboard/controllers/class.utilitycontroller.php#L336-L342 - which tries to preserve such additional structure made by plugins on application setup - is rendered useless.

Here the relevant code in Gdn_ApplicationManager::testApplication() which causes the issue:

```
...
// Call the application's setup method
$hooks = $applicationName.'Hooks';
if (!class_exists($hooks)) {
    $hooksPath = PATH_APPLICATIONS.DS.$ApplicationFolder.'/settings/class.hooks.php';
    if (file_exists($hooksPath)) {
        include_once $hooksPath;
    }
}
if (class_exists($hooks)) {
    /* @var Gdn_IPlugin $hooks The hooks object should be a plugin. */
    $hooks = new $hooks();
     if (method_exists($hooks, 'setup')) {
        $hooks->setup();
    }
}
...
```

This Issue was generated from your [forums](http://vanillaforums.org/discussion/comment/232207#Comment_232207)

**Comment by bleistivt** [comment]
Changing the Explicit parameter to false will fix this.
https://github.com/vanilla/vanilla/blob/e3b4649831a352656e5725ea528d83c4b2410981/applications/vanilla/settings/class.hooks.php#L763

But there is most likely a reason for it being explicitly set to true.

**Comment by linc** [comment]
That was resolved in 2.3 I believe.
